### PR TITLE
♻️ Migrate amp-cache example to amp.dev

### LINKED ISSUE #2627: AMP Cache Example: migrate ampbyexample link
> It is OK for URLs to include parameters in the query string, simply include these in the AMP Cache URL as well.
> 
> Example: https://ampbyexample-com.cdn.ampproject.org/c/s/ampbyexample.com/g?value=Hello%20World.

https://amp.dev/documentation/examples/guides/using_the_google_amp_cache/

### PR DESCRIPTION
**Pull request by lluerich**
Fixes #2627.

### CODE REVIEW

File: examples/source/2.guides/Using_the_Google_AMP_Cache/api.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const utils = require('@lib/utils');
+const {setMaxAge} = require('@lib/utils/cacheHelpers');
+const nunjucks = require('nunjucks');
+
+// eslint-disable-next-line new-cap
+const examples = express.Router();
+
+const filePath = utils.project.absolute('/examples/static/samples/files/get-example.html');
+
+examples.get('/g', (request, response) => {
```
Comment (sebastianbenz) [inline examples/source/2.guides/Using_the_Google_AMP_Cache/api.js]: can you rename this to `s/g/timestamp`

File: examples/source/2.guides/Using_the_Google_AMP_Cache/api.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const express = require('express');
+const utils = require('@lib/utils');
+const {setMaxAge} = require('@lib/utils/cacheHelpers');
+const nunjucks = require('nunjucks');
```
Comment (sebastianbenz) [inline examples/source/2.guides/Using_the_Google_AMP_Cache/api.js]: Can you please move this to a top-level example API. Thanks!

Comment (lluerich) [reply examples/source/2.guides/Using_the_Google_AMP_Cache/api.js]: Do you mean that it should be located within ```/examples/api```?

Comment (sebastianbenz) [reply examples/source/2.guides/Using_the_Google_AMP_Cache/api.js]: genau

File: examples/source/2.guides/Using_the_Google_AMP_Cache/api.js (hunk 1 above)
Comment (lluerich) [reply examples/source/2.guides/Using_the_Google_AMP_Cache/api.js]: Did you mean ```/s/g/timestamp``` or just ```/g/timestamp```?

Comment (lluerich) [reply examples/source/2.guides/Using_the_Google_AMP_Cache/api.js]: Went with ```/g/timestamp```. Please let me know, if it is supposed to be ```s/g/timestamp```.

File: examples/source/2.guides/Using_the_Google_AMP_Cache/api.js (hunk 2 above)
Comment (lluerich) [reply examples/source/2.guides/Using_the_Google_AMP_Cache/api.js]: PTAL

**Comment by lluerich** [comment]
Conflicts resolved, PTAL.

**Comment by sebastianbenz** [comment]
Thanks!

**Pull request merged**